File: Juan/P2/java/Napakalaki/src/napakalaki/Napakalaki.java

package napakalaki;

import java.util.ArrayList;

/**
 *
 * @author juane
 * @version 2.1
 * clase singleton
 */
public class Napakalaki {
    private static final Napakalaki instance = null;
    
    /**
     * Constructor privado
     */
    private Napakalaki(){
        
    }
    /**
     * Método initPlayers(ArrayList<String> names
     * @param names 
     */
    private void initPlayers(ArrayList<String> names){
        
    }
    
    /**
     * Método nextPlayer
     * @return 
     */
    private Player nextPlayer(){
        return null; //cambiar
    }
    
    /**
     * Método nextTurnAllowed()
     * @return 
     */
    private boolean nextTurnAllowed(){
        return false; //cambiar
    }
    
    /**
     * Método setEnemies()
     */
    private void setEnemies(){
        
    }
    
    
    /**
     * Método getInstance()
     * @return 
     */
    public static Napakalaki getInstance(){
        return instance;
    }
    
    /**
     * Método developCombat()
     * @return 
     */
    public CombatResult developCombat(){
        return null; //cambiar
    }
    
    /**
     * Método discardVisibleTreasures(ArrayList<Treasure> treasures)
     * @param treasures 
     */
    public void discardVisibleTreasures(ArrayList<Treasure> treasures){
        
    }
    
    /**
     * Método discardHiddenTreasures(ArrayList<Treasure> treasures)
     * @param treasures 
     */
    public void discardHiddenTreasures(ArrayList<Treasure> treasures){
        
    }
    
    /**
     * Método makeTreasuresVisible(ArrayList<Treasure> treasures)
     * @param treasures 
     */
    public void makeTreasuresVisible(ArrayList<Treasure> treasures){
        
    }
    
    /**
     * Método initGame(ArrayList<String> players)
     * @param players 
     */
    public void initGame(ArrayList<String> players){
        
    }
    
    /**
     * Método getCurrentPlayer()
     * @return 
     */
    public Player getCurrentPlayer(){
        return null; //cambiar
    }
    
    /**
     * Método getCurrentMonster()
     * @return 
     */
    
    public Monster getCurrentMonster(){
        return null;
    }
    
    /**
     * Método nextTurn()
     * @return 
     */
    public boolean nextTurn(){
        return false; // cambiar
    }
    
    /**
     * Método endOfGame(CombatResult result)
     * @param result
     * @return 
     */
    public boolean endOfGame(CombatResult result){
        return false; // cambiar
    }
}

File: Juan/P2_REVISION/Ruby/Napakalaki/pkg/Napakalaki-0.0.1/lib/Napakalaki.rb
include Singleton

class Napakalaki
    
    @@instance = null
#    Constructor

    def initialize(currentPlayer,players,dealer,currentMonster,unusedMonsters)
        @currentPlayer = currentPlayer
        @players = players
        @dealer = dealer
        @currentMonster = currentMonster
        @unusedMonsters = unusedMonsters
    end
    
    ## getCurrentPlayer
    ## getCurrentMonster
    attr_reader :currentPlayer
    attr_reader :currentMonster
    
    private
    def initPlayers(names)
        
    end
    
    private 
    def nextPlayer()
        
    end
    
    private
    def nextTurnAllowed()
        
    end
    
    private
    def setEnemies()
        
    end
    
    private 
    def self.getInstance()
        @@instance = Napakalaki.instance
    end
    
    def devlopCaombat()
        
    end
    
    def discardVisibleTreasures (treasures)
        
    end
    
    def discarHiddenTreasures(treasures)
        
    end
    
    def makeTreasuresVisible(treasures)
        
    end
    
    def initGame(players)
        
    end
    
    
    def nextTurn()
        
    end
    
    def endOfGame (result)
        
    end
    
end
